internal/incoming/odfi/download.go

```diff
@@ -70,25 +74,32 @@ func (d *downloadedFiles) deleteFiles() error {
 	return os.RemoveAll(d.dir)
 }
 
-func (d *downloadedFiles) deleteEmptyDirs(agent upload.Agent) error {
-	count := func(path string) int {
+func (d *downloadedFiles) deleteEmptyDirs(ctx context.Context, agent upload.Agent) error {
+	count := func(ctx context.Context, path string) int {
 		infos, err := os.ReadDir(path)
 		if err != nil {
 			return -1
 		}
+
+		_, span := telemetry.StartSpan(ctx, "odfi-delete-empty-dirs", trace.WithAttributes(
```

Comment:
Why don't we keep the context here?

Reply:
There's no further calls to pass the context into. We just need the span to record timing information for how long the function takes.